docs/faq/events.md

```diff
@@ -28,3 +30,182 @@ There are lots of times when words just aren't enough to really get clarity on a
+
+In some cases you will need to add event listener to elements outside of the element you are building: e.g. a `resize` listener of the `window`. In this case the `window` could last well beyond the last referenc to your element and it is very important that you do the work to ensure that the listener is added and removed appropriately. Thankfully, the custom elements specification outlines the [`connectedCallback` and `disconnectedCallback` lifecycle methods](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#Using_the_lifecycle_callbacks). These are the perfect hooks for managing these sorts of listeners. When adding event listeners to this sort of external element, use the `connectedCallback`, like so:
+
+```js
+connectedCallback() {
+    super.connectedCallback && super.connectedCallback();
+    window.addEventListener('resize', this.handleResize);
+}
+handleResize = (event) => {
+    // Do something with the resize event here.
+}
+```
+
+Ensuring that event listener does not stick around beyond the presence of your element in the DOM is just as straight forward with the `disconnectedCallback` method.
+
+```js
+disconnectedCallback() {
+    window.removeEventListener('resize', this.handleResize);
+    super.disconnectedCallback && super.disconnectedCallback();
+}
+```
+
+Side note: again, as you will be adding this listener manually, you _will_ need to ensure that `this` is bound to your liking. See the class field method syntax above. When removing event listeners is it also important to ensure that the identity of the callback bound to the event is available by reference or it won't be propertly removed.
+
+## Event dispatching
+
+### Keep it simple
+
+You can dispatch a DOM event on any element in your applicaiton with `el.dispatchEvent(new Event('event'))`. With just that amount of code you'll be able to trigger the callback for any listener for that event name that is bound to that element. You'll also trigger the callback on any ancestor elements in the same DOM tree for listeners bound to the "capture" phase of an event. In a lot of cases this might be all of the functionality you really need when dispatching an event, so compare your usage needs before worrying too much about the following.
+
+### Know your options
+
+If you want to be able to listen for the event on ancestor elements during the "bubble" phase you'll need to manage the `bubbles` property of the event, like so:
+
+```js
+el.dispatchEvent(
+  new Event('event', {
+    bubbles: true,
+  }),
+);
+```
+
+This will allow your event to "bubble" up to the top of the current DOM tree. `bubbles` isn't required in order for an event to be accessible to the ancestors of the element on which is it dispatched, but in that listening to events on the "capture" phase can sometimes be confusing, involving of extended techniques to do in declarative templating, and requiring of extended documentation, it is best that you do use it in these cases.
```

Comment:
this second sentence is really long and confusing, starting at "`bubbles` isn't required .... " to "do use it in these cases". Could you reword a bit and make it multiple sentence instead of one long sentence?